design/one-pager-mrds.md

```diff
@@ -0,0 +1,248 @@
+create its MRDs with `spec.desiredState: Active`. This'll cause Crossplane's MRD
+controller to immediately create the necessary CRDs.
+
+## Migration Plan
+
+This proposal should be relatively simple to implement in Crossplane core, but
+it requires significant changes for providers.
+
+Today providers use [controller-tools][6] to generate YAML CRD manifests from Go
+types. These CRDs are then baked into a Crossplane OCI package using the
+`crossplane` CLI.
+
+For a provider to package MRDs instead of CRDs it'd need a tool that can
+generate MRD YAML manifests from Go structs. This could be done 'directly' or by
+generating a CRD and post-processing it to create an MRD - e.g. changing the
+type and adding the connection details.
+
+Older versions of Crossplane w
+ouldn't understand MRDs, so a provider that
+switches from CRDs to MRDs wouldn't work on older versions of Crossplane.
+
+To alleviate these migration issues I propose that Crossplane automatically
+translate MR CRDs to MRDs at package install time. Providers would be expected
+to package MRDs directly, but this could happen once all supported versions of
+Crossplane had GA support for MRDs.
+
+## Alternatives Considered
+
+I considered the following alternatives before landing on this proposal.
+
+### Use CRD Annotations for Connection Details
+
+Connection details (not CRD filtering) were the original motivation for the MRD
+design. I wasn't convinced MRDs were worth the churn purely as a place to
+document connection details. The alternative to MRDs for this use case was to
+annotate CRDs, e.g.:
+
+```yaml
+metadata:
+  annotations:
+    connection.crossplane.io/username: The username for this MR
+    connection.crossplane.io/address: The IP address for this MR
+```
+
+This'd likely be a good enough solution for connection details, but it's a worse
+UX compared to MRD due to the lack of schema.
+
+Given MRDs are also useful for CRD filtering, I discarded this alternative.
+
+### Use `spec.enabledAPIs` to Filter CRDs
+
+When we first started discussing filtering CRDs, adding `spec.enabledAPIs` to
+the Provider package was a popular idea. For example:
+
+```yaml
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: crossplane-provider-tf-azure
+spec:
+  enabledAPIs:
+  - Provider.*
+  - virtual.*
+  - lb
+  - resource.azure.tf.crossplane.io/v1alpha1.ResourceGroup
+  package: ulucinar/provider-tf-azure-arm64:build-83b9bfc0
+```
+
+This would be passed down to the ProviderRevision and ultimately to the running
+provider binary by configuring an `--enabled-apis` flag.
+
+This seems like a good UX on the surface, but dependencies make it challenging.
+We'd need to update the package dependency model to be aware of what APIs a
+dependent Configuration needed - not just what providers it needed. If multiple
+Configurations depended on the same provider Crossplane would need to compute
+the set of APIs (MRs) that should be enabled.
```

Comment:
The package manager would have to read and compute the dag, and then trim the tree based on what is selected and install everything that is left.